### PR TITLE
Added support for different mysql ports + using correct constants

### LINKED ISSUE #440: Backups do not work when using different MySQL ports
When using a different MySQL database port but the default one, the backup app will not work.

At the creation of the database dump, the database port is correctly fetched:
https://github.com/nextcloud/backup/blob/0b02c80d388d00c94da49fbff4df6884a51abb7e/lib/Service/PointService.php#L552-L564
https://github.com/nextcloud/backup/blob/0b02c80d388d00c94da49fbff4df6884a51abb7e/lib/Service/PointService.php#L528-L528

But is omitted in the final dump call so the default port will be used:
https://github.com/nextcloud/backup/blob/0b02c80d388d00c94da49fbff4df6884a51abb7e/lib/SqlDump/SqlDumpMySQL.php#L69-L70

This causes the backup creation process to fail.

### PR DESCRIPTION
**Pull request by ksoeren**
fixes #440

### CODE REVIEW

**Review by ArtificialOwl — APPROVED** [review]

**Review by ArtificialOwl — APPROVED** [review]

File: lib/SqlDump/SqlDumpMySQL.php [hunk 1]
```diff
@@ -67,7 +67,10 @@ public function __construct() {
 	 * @throws SqlDumpException
 	 */
 	public function export(array $params, string $filename): void {
-		$connect = 'mysql:host=' . $params['dbhost'] . ';dbname=' . $params['dbname'];
+		$connect = sprintf('mysql:host=%s;dbname=%s', $params[ISqlDump::DB_HOST], $params[ISqlDump::DB_NAME]);
+		if(false === empty($params[ISqlDump::DB_PORT])){
```
Comment (michitux) [inline lib/SqlDump/SqlDumpMySQL.php:71]: To fix the code style violation:

```suggestion
		if (false === empty($params[ISqlDump::DB_PORT])) {
```

**Comment by kintaro1981** [comment]
is this going to be in the next release?

**Comment by dacog** [comment]
Hi @michitux  Is there anything still missing for this to be merged?  Can we help somehow?

**Comment by michitux** [comment]
> Hi @michitux  Is there anything still missing for this to be merged?  Can we help somehow?

I'm not a maintainer of this extension or in another way affiliated with Nextcloud, I just tried to help fixing the code style violation as the automated checks were failing.

**Comment by dacog** [comment]
> > Hi @michitux  Is there anything still missing for this to be merged?  Can we help somehow?
> 
> I'm not a maintainer of this extension or in another way affiliated with Nextcloud, I just tried to help fixing the code style violation as the automated checks were failing.

Hi @michitux  Sorry! I did not know that. 

@ArtificialOwl are you the mantainer for this app?

**Comment by chrfritsch** [comment]
@artonge Can you please help here?

**Pull request merged**